lib.rs

```diff
@@ -93,3 +93,846 @@ pub mod src {
+        }
+
+        pub fn get_strict_std_compliance(&self) -> bool {
+            self.dav1d_settings.strict_std_compliance != 0
+        }
+
+        pub fn set_output_invisible_frames(&mut self, output_invisible_frames: bool) {
+            self.dav1d_settings.output_invisible_frames = i32::from(output_invisible_frames);
+        }
+
+        pub fn get_output_invisible_frames(&self) -> bool {
+            self.dav1d_settings.output_invisible_frames != 0
+        }
+
+        pub fn set_inloop_filters(&mut self, inloop_filters: InloopFilterType) {
+            self.dav1d_settings.inloop_filters = inloop_filters.bits();
+        }
+
+        pub fn get_inloop_filters(&self) -> InloopFilterType {
+            InloopFilterType::from_bits_truncate(self.dav1d_settings.inloop_filters)
+        }
+
+        pub fn set_decode_frame_type(&mut self, decode_frame_type: DecodeFrameType) {
+            self.dav1d_settings.decode_frame_type = decode_frame_type.into();
+        }
+
+        pub fn get_decode_frame_type(&self) -> DecodeFrameType {
+            DecodeFrameType::try_from(self.dav1d_settings.decode_frame_type)
+                .expect("Invalid Dav1dDecodeFrameType")
+        }
+    }
+
+    bitflags::bitflags! {
+        #[derive(Copy, Clone, Debug, PartialEq, Eq, Default)]
+        pub struct InloopFilterType: u32 {
+            const DEBLOCK = DAV1D_INLOOPFILTER_DEBLOCK;
+            const CDEF = DAV1D_INLOOPFILTER_CDEF;
+            const RESTORATION = DAV1D_INLOOPFILTER_RESTORATION;
+        }
+    }
+
+    #[derive(Default, Debug, Copy, Clone, PartialEq, Eq)]
+    pub enum DecodeFrameType {
+        #[default]
+        All,
+        Reference,
+        Intra,
+        Key,
+    }
+
+    impl TryFrom<u32> for DecodeFrameType {
+        type Error = TryFromEnumError;
+
+        fn try_from(value: u32) -> Result<Self, Self::Error> {
+            match value {
+                DAV1D_DECODEFRAMETYPE_ALL => Ok(DecodeFrameType::All),
+                DAV1D_DECODEFRAMETYPE_REFERENCE => Ok(DecodeFrameType::Reference),
+                DAV1D_DECODEFRAMETYPE_INTRA => Ok(DecodeFrameType::Intra),
+                DAV1D_DECODEFRAMETYPE_KEY => Ok(DecodeFrameType::Key),
+                _ => Err(TryFromEnumError(())),
+            }
+        }
+    }
+
+    impl From<DecodeFrameType> for u32 {
+        fn from(v: DecodeFrameType) -> u32 {
+            match v {
+                DecodeFrameType::All => DAV1D_DECODEFRAMETYPE_ALL,
+                DecodeFrameType::Reference => DAV1D_DECODEFRAMETYPE_REFERENCE,
+                DecodeFrameType::Intra => DAV1D_DECODEFRAMETYPE_INTRA,
+                DecodeFrameType::Key => DAV1D_DECODEFRAMETYPE_KEY,
+            }
+        }
+    }
+
+    /// The error type returned when a conversion from a C enum fails.
+    #[derive(Debug, Copy, Clone, PartialEq, Eq)]
+    pub struct TryFromEnumError(());
+
+    impl std::fmt::Display for TryFromEnumError {
```

Comment:
We've preferred fully importing these kinds of items (e.x. `use std::fmt::Display`).